exercism_test_helper/test/meta/test_parser_test.exs

```diff
@@ -0,0 +1,178 @@
+                   name: "test hello says 'Hello, World!'",
+                   task_id: nil,
+                   test_code: "assert HelloWorld.hello() == \"Hello, World!\""
+                 },
+                 %T{
+                   name: "test hello says 'Hello, Alice!'",
+                   task_id: nil,
+                   test_code: "assert HelloWorld.hello(\"Alice\") == \"Hello, Alice!\""
+                 }
+               ]
+             }
+    end
+
+    test "three tests in two describes" do
+      code = """
+      defmodule(HelloWorldTest) do
+        use(ExUnit.Case)
+
+        describe "hello" do
+          test("says 'Hello, World!'") do
+            assert(HelloWorld.hello() == "Hello, World!")
+          end
+
+          test("says 'Hello, Alice!'") do
+            assert(HelloWorld.hello("Alice") == "Hello, Alice!")
+          end
+        end
+
+        describe "bye" do
+          test("says 'Bye :(") do
+            assert(HelloWorld.bye() == "Bye :(")
+          end
+        end
+      end
+      """
+
+      assert Meta.TestParser.parse(code) == %Meta.TestParser.TestSuite{
+               description: nil,
+               task_id: nil,
+               tests: [
+                 %T{
+                   name: "test hello says 'Hello, World!'",
+                   task_id: nil,
+                   test_code: "assert HelloWorld.hello() == \"Hello, World!\""
+                 },
+                 %T{
+                   name: "test hello says 'Hello, Alice!'",
+                   task_id: nil,
+                   test_code: "assert HelloWorld.hello(\"Alice\") == \"Hello, Alice!\""
+                 },
+                 %T{
+                   name: "test bye says 'Bye :(",
+                   task_id: nil,
+                   test_code: "assert HelloWorld.bye() == \"Bye :(\""
+                 }
+               ]
+             }
+    end
+
+    test "task ids - optional, only apply to a single test" do
+      code = """
+      defmodule(HelloWorldTest) do
+        use(ExUnit.Case)
+
+        @task_id 1
+        test("test A1 - task 1") do
+        end
+
+        test("test A2 - general") do
+        end
+
+        describe "describe1" do
+          test("test B1 - general") do
+          end
+
+          @task_id 2
+          @task_id 1
+          test("test B2 - task 1") do
+          end
+
+          @task_id 1
+          test("test B3 - task 1") do
+          end
+
+          @task_id 2
+          test("test B4 - task 2") do
+          end
+
+          test("test B5 - general") do
+          end
+        end
+
+        describe "describe2" do
+          @task_id 3
+          test("test C1 - task 3") do
+          end
+
+          test("test C2 - general") do
+          end
+
+          @task_id 4
+          test("test C3 - task 4") do
+          end
+        end
+      end
+      """
+
+      assert Meta.TestParser.parse(code) == %Meta.TestParser.TestSuite{
+               description: nil,
+               task_id: nil,
```

Comment:
Should there be a task_id at the top level?

Reply:
Good point - no, that doesn't make sense. But I had to add it to the `Meta.TestParser.TestSuite` struct because it's used at the accumulator in `Meta.TestParser.(pre|post)_order_parse`. What would you advise to get around that? 🤔

Reply:
let me pull it local, I think your solution is very close

Reply:
Disregard my comment, I forgot how this worked until looking at this.  We are using `TestSuite` as an accumulator and when we are ejecting the `%TestSuite{}` as json using `Jason` it will disregard that struct field and shouldn't show up in the json file at all.

One way to assert this would be to write a jq test to assert the final json result in the smoke test workflow

Reply:
https://stackoverflow.com/a/31933234/8583288 is an interesting discussion of asserting json files are equal using jq